### ISSUE TITLE
Declaration JSDoc doesn't work with destructuring.

### ISSUE THREAD

**Issue by brad4d** [google/closure-compiler#2067]
```javascript
const values = ['a', 1];
const /** string */ s = values[0];
const /** number */ n = values[1];

const [ /** string */ x, /** number */ y] = values;
```
```
JSC_MISPLACED_ANNOTATION: Misplaced type annotation. Type annotations are not allowed here. Are you missing parentheses? at line 5 character 22
const [ /** string */ x, /** number */ y] = values;
                      ^
JSC_MISPLACED_ANNOTATION: Misplaced type annotation. Type annotations are not allowed here. Are you missing parentheses? at line 5 character 39
const [ /** string */ x, /** number */ y] = values;
                                       ^
```

https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520SIMPLE_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250Aconst%2520values%2520%253D%2520%255B'a'%252C%25201%255D%253B%250Aconst%2520%252F**%2520string%2520*%252F%2520s%2520%253D%2520values%255B0%255D%253B%250Aconst%2520%252F**%2520number%2520*%252F%2520n%2520%253D%2520values%255B1%255D%253B%250A%250Aconst%2520%255B%2520%252F**%2520string%2520*%252F%2520x%252C%2520%252F**%2520number%2520*%252F%2520y%255D%2520%253D%2520values%253B%250A

Behaviour is the same with the `/** @type {} */` form.

**Comment by dimvar** [comment]
Weird. It gives the warning but then actually uses the type annotation.

https://closure-compiler-debugger.appspot.com/#input0%3Dconst%2520values%2520%253D%2520%255B'a'%252C%25201%255D%253B%250Aconst%2520%255B%2520%252F**%2520string%2520*%252F%2520x%252C%2520%252F**%2520number%2520*%252F%2520y%255D%2520%253D%2520values%253B%250Avar%2520%252F**%2520null%2520*%252F%2520foobar%2520%253D%2520x%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by brad4d** [comment]
No, it doesn't use the type annnotation. I can reassign a variable created through destructuring to an incorrect type without warning.

```javascript
const values = ['a', 1];
let [ /** string */ x, /** number */ y] = values;
x = 5;
```

https://closure-compiler-debugger.appspot.com/#input0%3Dconst%2520values%2520%253D%2520%255B'a'%252C%25201%255D%253B%250Alet%2520%255B%2520%252F**%2520string%2520*%252F%2520x%252C%2520%252F**%2520number%2520*%252F%2520y%255D%2520%253D%2520values%253B%250Ax%2520%253D%25205%253B%250A%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1
